src/Zoo.cpp

```diff
@@ -0,0 +1,267 @@
+#include "Zoo.h"
+#include "Lion.h"
+#include "Monkey.h"
+#include "Shark.h"
+#include "Animal.h"
+#include <random_util.h>
+#include <stdexcept>
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+
+#define MAP_ROWS 20
+#define MAP_COLUMNS 40
+
+/**
+* @brief parses the command and arguments from the CLI.
+* @return std::vector<std::string> containing splited command and arguments.
+**/
+std::vector<std::string> parse_command()
+{
+	std::string line;
+	std::getline(std::cin, line);  
+
+
+	std::istringstream iss(line);  // create an istringstream object from the line
+	std::vector<std::string> tokens;  
+	if (!line.empty()) {
+
+		// read each whitespace-separated token into the vector
+		std::string token;
+		while (iss >> token) {
+			tokens.push_back(token);
+		}
+
+	}
+	return tokens;
+
+}
+
+void Zoo::stop(const int index)
+{
+	if (index >= animals.size())
+		throw std::out_of_range("Index is out of range.\n");
+	animals[index]->stop();
+}
+void Zoo::move(const int index)
+{
+	if (index >= animals.size())
+		throw std::out_of_range("Index is out of range.\n");
+	animals[index]->move();
+}
+void Zoo::create(const std::string animal_type, const std::string animal_name)
+{
+	if (animal_type != "Lion" && animal_type != "Monkey" && animal_type != "Shark")
+		throw std::invalid_argument("Animal type must be either Lion, Monkey, or Shark.\n");
+	Location l = Location(RandomUtil::generateRandomValue(0, 19), RandomUtil::generateRandomValue(0, 39));
```

Comment:
you can use consts for bounderies it's more effecient if you to change the map size